worker/online_restore.go

```diff
@@ -336,15 +383,28 @@ func handleRestoreProposal(ctx context.Context, req *pb.RestoreRequest) error {
 		return errors.Wrapf(err, "cannot load schema after restore")
 	}
 
-	// reset gql schema
+	// Reset gql schema only when the restore is not partial, so that after this restore the cluster
+	// can be in non-draining mode and hence gqlSchema can be lazy loaded.
 	glog.Info("reseting local gql schema store")
-	ResetGQLSchemaStore()
+	if !req.IsPartial {
```

Comment:
What will happen if we reset the schema even if the restore was partial?

Reply:
We'll not be able to send restore request, as all the admin resolvers would have become nil and they can't be loaded as well because cluster is in draining mode.